Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.swift

```diff
@@ -0,0 +1,254 @@
+                scene.baseSurface.opacity = isPresented ? 0.6 : 0
+            }
+            .task {
+                try? await locationDataSource.start()
+                
+                for await location in locationDataSource.locations {
+                    try? await routeDataModel.routeTracker?.track(location)
+                }
+            }
+            .overlay(alignment: .top) {
+                Text(statusText)
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button("Start") {
+                        isNavigating = true
+                        Task {
+                            do {
+                                try await startNavigation()
+                            } catch {
+                                print("Failed to start navigation.")
+                            }
+                        }
+                    }
+                    .disabled(isNavigating)
+                }
+            }
+            .onDisappear {
+                Task { await locationDataSource.stop() }
+            }
+        }
+    }
+    
+    /// Create a graphic overlay and adds a graphic (with solid yellow 3D tube symbol)
+    /// to represent the route.
+    @MainActor
+    private func makeRouteOverlay(routeResult: RouteResult, routeGraphic: Graphic) -> GraphicsOverlay {
+        let graphicsOverlay = GraphicsOverlay()
+        graphicsOverlay.sceneProperties.surfacePlacement = .absolute
+        let strokeSymbolLayer = SolidStrokeSymbolLayer(
+            width: 1.0,
+            color: .yellow,
+            lineStyle3D: .tube
+        )
+        let polylineSymbol = MultilayerPolylineSymbol(symbolLayers: [strokeSymbolLayer])
+        let polylineRenderer = SimpleRenderer(symbol: polylineSymbol)
+        graphicsOverlay.renderer = polylineRenderer
+        
+        if let originalPolyline = routeResult.routes.first?.geometry {
+            addingElevation(to: originalPolyline) { polyline in
+                routeGraphic.geometry = polyline
+                graphicsOverlay.addGraphic(routeGraphic)
+            }
+        }
+        
+        return graphicsOverlay
+    }
+    
+    /// Densify the polyline geometry so the elevation can be adjusted every 0.3 meters,
+    /// and add an elevation to the geometry.
+    ///
+    /// - Parameters:
+    ///   - polyline: The polyline geometry of the route.
+    ///   - z: A `Double` value representing z elevation.
+    ///   - completion: A completion closure to execute after the polyline is generated with success or not.
+    private func addingElevation(
+        to polyline: Polyline,
+        elevation z: Double = 3,
```

Comment:
Since this is only called in one place, I would suggest moving the default value to where it is called and then moving the `elevation` parameter to the front, without a label:

```swift
private func addingElevation(_ elevation z: Double, ...)
```